flink-runtime/src/test/java/org/apache/flink/runtime/deployment/ShuffleDescriptorTest.java

```diff
@@ -0,0 +1,187 @@
+				state == ExecutionState.DEPLOYING) {
+				NettyShuffleDescriptor nettyShuffleDescriptor;
+
+				// Create local or remote channels
+				assertThat(localShuffleDescriptor, instanceOf(NettyShuffleDescriptor.class));
+				assertThat(localShuffleDescriptor.isUnknown(), is(false));
+				nettyShuffleDescriptor = (NettyShuffleDescriptor) localShuffleDescriptor;
+				assertThat(nettyShuffleDescriptor.getResultPartitionID(), is(localPartitionId));
+				assertThat(nettyShuffleDescriptor.isLocalTo(consumerResourceID), is(true));
+
+				assertThat(remoteShuffleDescriptor, instanceOf(NettyShuffleDescriptor.class));
+				assertThat(remoteShuffleDescriptor.isUnknown(), is(false));
+				nettyShuffleDescriptor = (NettyShuffleDescriptor) remoteShuffleDescriptor;
+				assertThat(nettyShuffleDescriptor.getResultPartitionID(), is(remotePartitionId));
+				assertThat(nettyShuffleDescriptor.isLocalTo(consumerResourceID), is(false));
+				assertThat(nettyShuffleDescriptor.getConnectionId(), is(CONNECTION));
+			} else {
+				// Unknown (lazy deployment allowed)
+				assertThat(localShuffleDescriptor, instanceOf(UnknownShuffleDescriptor.class));
+				assertThat(localShuffleDescriptor.isUnknown(), is(true));
+				assertThat(localShuffleDescriptor.getResultPartitionID(), is(localPartitionId));
+
+				assertThat(remoteShuffleDescriptor, instanceOf(UnknownShuffleDescriptor.class));
+				assertThat(remoteShuffleDescriptor.isUnknown(), is(true));
+				assertThat(remoteShuffleDescriptor.getResultPartitionID(), is(remotePartitionId));
+			}
+
+			assertThat(unknownShuffleDescriptor, instanceOf(UnknownShuffleDescriptor.class));
+			assertThat(unknownShuffleDescriptor.isUnknown(), is(true));
+			assertThat(unknownShuffleDescriptor.getResultPartitionID(), is(unknownPartitionId));
+		}
+	}
+
+	@Test
+	public void testUnknownChannelWithoutLazyDeploymentThrows() throws Exception {
+		ResultPartitionID unknownPartitionId = new ResultPartitionID();
+
+		// This should work if lazy deployment is allowed
+		ShuffleDescriptor unknownSdd = getConsumedPartitionShuffleDescriptor(
+			unknownPartitionId,
+			ExecutionState.CREATED,
+			null,
+			true);
+
+		assertThat(unknownSdd, instanceOf(UnknownShuffleDescriptor.class));
+		assertThat(unknownSdd.isUnknown(), is(true));
+		assertThat(unknownSdd.getResultPartitionID(), is(unknownPartitionId));
+
+		try {
+			// Fail if lazy deployment is *not* allowed
+			getConsumedPartitionShuffleDescriptor(
+				unknownPartitionId,
+				ExecutionState.CREATED,
+				null,
+				false);
+			fail("Did not throw expected ExecutionGraphException");
+		} catch (ExecutionGraphException ignored) {
+		}
+	}
+
+	private static ShuffleDescriptor getConsumedPartitionShuffleDescriptor(
+			ResultPartitionID id,
+			ExecutionState state,
+			@Nullable ResultPartitionDeploymentDescriptor producedPartition,
+			boolean allowLazyDeployment) throws ExecutionGraphException {
+		return TaskDeploymentDescriptorFactory.getConsumedPartitionShuffleDescriptor(
```

Comment:
We could verify `assertThat(SDD.getResultPartitionID(), is(id))` before return for all the above tests.